packages/wallet/api/src/lib-wallet.js

```diff
@@ -1528,6 +1528,34 @@ export function makeWallet({
     return offerResult.uiNotifier;
   }
 
+  /**
+   * Gets the public notifiers from an offer's result.
+   *
+   * @param {string} rawId - The offer's raw id.
+   * @param {string} dappOrigin - The origin of the dapp the offer came from.
+   */
+  async function getPublicNotifiers(rawId, dappOrigin = 'unknown') {
+    const id = makeId(dappOrigin, rawId);
+
+    const offerResult = await idToOfferResultPromiseKit.get(id).promise;
+    assert(
+      passStyleOf(offerResult) === 'copyRecord',
+      `offerResult ${offerResult} must be a record to have publicNotifiers`,
+    );
```

Comment:
Is this exceptional? Reading this, I take it an offerResult can be other things if it doesn't have publicNotifiers and doesn't necessarily have publicNotifiers.

The asserts in this function treat it as an error to call `getPublicNotifiers` for an offer that doesn't have publicNotifiers. That would be surprising to me as a user of the function as I expect generally calling getThings when there are no things to just return a value indicating that: null, undefined or an empty collection. In this case, it may help to distinguish no key from a value that is set so I'd suggest returning `undefined` instead of these first two throws. (The third looks like a necessary type guard.)

If the throws behavior of this function is better for how dapps are programmed, I'd like to understand that better and also request that `@throws` be used in the jsdoc to make it explicit.

Reply:
I think for contracts, a successful offer result should either always have a notifier or never have one. If the existence of a notifier is being used as a signal for something in the dapp, there's probably better ways to infer whatever that signal is.

That assumption would boil this case down to just programming error--you either implemented your contract incorrectly or tried to access a notifier on the wrong type of offer result. In that case, I think seeing these errors from a capTP call in the browser is more helpful for debugging.

Reply:
> I think for contracts, a successful offer result should either always have a notifier or never have one.

Agreed.

> That assumption would boil this case down to just programming error--you either implemented your contract incorrectly or tried to access a notifier on the wrong type of offer result.

It's that latter case I'm asking about. It sounds like you're saying that before you call `getPublicNotifiers` you should know whether the offer result has public notifiers. If that's so I think the jsdoc should indicate that somehow. E.g. `@throws if the offer result doesn't have notifiers`.

Reply:
Added `@throws ...` to the jsdoc